wallet/tests/data_requests.rs

```diff
@@ -63,34 +59,93 @@ fn test_radon_error_json_serialization() {
     assert_eq!(serde_json::to_string(&radon_error).unwrap(), expected_json);
 }
 
-#[test]
 /// This is a rather end-2-end test that applies a script on some JSON input and checks whether the
 /// final `RadonReport` complies with the Witnet Wallet API.
-fn test_data_request_report_json_serialization() {
-    let input = RadonTypes::from(RadonString::from(
-        r#"{"coord":{"lon":13.41,"lat":52.52},"weather":[{"id":600,"main":"Snow","description":"light snow","icon":"13n"}],"base":"stations","main":{"temp":-4,"pressure":1013,"humidity":73,"temp_min":-4,"temp_max":-4},"visibility":10000,"wind":{"speed":2.6,"deg":90},"clouds":{"all":75},"dt":1548346800,"sys":{"type":1,"id":1275,"message":0.0038,"country":"DE","sunrise":1548313160,"sunset":1548344298},"id":2950159,"name":"Berlin","cod":200}"#,
-    ));
-    let script = vec![
-        (RadonOpCodes::StringParseJSONMap, None),
-        (
-            RadonOpCodes::MapGetMap,
-            Some(vec![Value::Text(String::from("main"))]),
-        ),
-        (
-            RadonOpCodes::MapGetFloat,
-            Some(vec![Value::Text(String::from("temp"))]),
-        ),
-    ];
-    let mut context = ReportContext::default();
-    let report = execute_radon_script(
-        input,
-        &script,
-        &mut context,
-        RadonScriptExecutionSettings::enable_all(),
-    )
-    .unwrap();
-
-    let json = serde_json::ser::to_string(&report).unwrap();
-
-    assert_eq!(json, "");
+#[actix_rt::test]
+#[cfg(feature = "side_effected")]
+async fn test_data_request_report_json_serialization() {
```

Comment:
We have similar tests in `node/tests/data_request_examples.rs`. They avoid side effects by storing the result of the URL locally. Perhaps we can integrate this test with the other ones.

Reply:
This one is explicitly testing the `try_data_request_locally` method, which provides no option for operating on static data, and directly calls the `run_retrieval_report` function that has the same issue.

Reply:
If it really can't be avoided, I would prefer to run a local webserver and handle the data requests using that, rather than querying external APIs.

Reply:
Which one do you prefer from these options for moving forward with this?
- leaving it like it is, provided that `side_effected` is not a default option and thus this test will only be run if explicitly asked for.
- making `try_data_request_locally` and `run_retrieval_report` accept data injection through its signature.
- removing this test altogether.

Reply:
A test that is only run if explicitly asked for is useless, we will merge changes that break that test and not realize it.

> * making try_data_request_locally and run_retrieval_report accept data injection through its signature.

This sounds cool.

> removing this test altogether.

This also sounds good as long as there is a test somewhere that serializes a `RadonReport` to JSON, the `RadonReport` can be mocked or generated using a different function but we need to make sure that the API does not change.

Reply:
Done, at last!